src/uct/ib/base/ib_device.c

```diff
@@ -527,21 +529,29 @@ ucs_status_t uct_ib_device_query(uct_ib_device_t *dev,
         break;
     }
 
-    if (dev->num_ports > UCT_IB_DEV_MAX_PORTS) {
-        ucs_error("%s has %d ports, but only up to %d are supported",
-                  ibv_get_device_name(ibv_device), dev->num_ports,
-                  UCT_IB_DEV_MAX_PORTS);
-        return UCS_ERR_UNSUPPORTED;
-    }
-
+    num_ports = dev->num_ports < UCT_IB_DEV_MAX_PORTS ? dev->num_ports : UCT_IB_DEV_MAX_PORTS;
     /* Query all ports */
-    for (i = 0; i < dev->num_ports; ++i) {
+    for (i = 0; i < num_ports; ++i) {
         ret = ibv_query_port(dev->ibv_context, i + dev->first_port,
                              &dev->port_attr[i]);
         if (ret != 0) {
             ucs_error("ibv_query_port() returned %d: %m", ret);
             return UCS_ERR_IO_ERROR;
         }
+        if (dev->port_attr[i].gid_tbl_len > 0) {
+            has_gid = 1;
+            break;
+        }
```

Comment:
```
if (dev->port_attr[i].gid_tbl_len == 0) {
    return UCS_ERR_UNSUPPORTED;
}
```

Reply:
@yosefe One device has max 2 ports are allowed. If any one port with gid_tbl_len == 0 then we think the device is UNSUPPORTED even if the other port of the same device with gid_tbl_len > 0. Right? Maybe we won't have this case?

Reply:
we can check num of gid in uct_ib_device_port_check() to enable each port separately